dbms/src/Flash/Mpp/ReceivedMessageQueue.cpp

```diff
@@ -152,6 +150,15 @@ MPMCQueueResult ReceivedMessageQueue::pop(size_t stream_id, ReceivedMessagePtr &
                 grpc_recv_queue.tryDequeue();
 #endif
             }
+            ExchangeReceiverMetric::subDataSizeMetric(*data_size_in_queue, recv_msg->getPacket().ByteSizeLong());
```

Comment:
Should be a bug in the previous implementation, this `subDataSizeMetric` should only be called if `recv_msg->getRemainingConsumers()->fetch_sub(1) == 1`

Reply:
ok, updated.